pallets/compliance-manager/src/lib.rs

```diff
@@ -301,26 +355,42 @@ decl_module! {
         #[weight = T::DbWeight::get().reads_writes(1, 1) + 600_000_000 + 1_000_000 * u64::try_from(max(sender_conditions.len(), receiver_conditions.len())).unwrap_or_default()]
         pub fn add_compliance_requirement(origin, ticker: Ticker, sender_conditions: Vec<Condition>, receiver_conditions: Vec<Condition>) {
             let did = Self::ensure_can_modify_rules(origin, ticker)?;
+            ensure!(
+                sender_conditions.iter().all(|cond| cond.issuers.len() <= T::MaxTrustedIssuerPerCondition::get()),
+                Error::<T>::MaxTrustedIssuerPerConditionExcedeed);
+            ensure!(
+                receiver_conditions.iter().all(|cond| cond.issuers.len() <= T::MaxTrustedIssuerPerCondition::get()),
+                Error::<T>::MaxTrustedIssuerPerConditionExcedeed);
```

Comment:
I don't think we need to limit individual conditions when we have an overall complexity limit on compliance requirements.